### PR TITLE
gh-118773: Use language-invariant SDDL string instead of aliases for ACLs.

### PR DESCRIPTION
**Pull request by zooba**
* Issue: gh-118773

### CODE REVIEW

**Comment by zooba** [comment]
!buildbot .*windows.*

**Comment by zooba** [comment]
FYI, this will be backported to all branches, but because the original change hasn't been merged into those yet it can't be done automatically. I'll handle it myself once this one is reviewed and merged.

**Comment by zooba** [comment]
The `test_cext` failure on ARM64 is unrelated.

File: Modules/posixmodule.c [hunk 1]
```diff
@@ -5727,26 +5587,31 @@ os_mkdir_impl(PyObject *module, path_t *path, int mode, int dir_fd)
...
-        error = initializeDefaultSecurityAttributes(&pSecAttr, &secAttrData);
-        break;
+    if (mode == 0700 /* 0o700 */) {
+        ULONG sdSize;
+        pSecAttr = &secAttr;
+        // Set a discreationary ACL (D) that is protected (P) and includes
+        // inheritable (OICI) entries that allow (A) full control (FA) to
+        // SYSTEM (SY), Administrators (BA), and the owner (OW).
+        if (!ConvertStringSecurityDescriptorToSecurityDescriptorW(
+            L"D:PAI(A;OICI;FA;;;SY)(A;OICI;FA;;;BA)(A;OICI;FA;;;OW)",
```
Comment (eryksun) [inline Modules/posixmodule.c]:
```suggestion
            L"D:P(A;OICI;FA;;;SY)(A;OICI;FA;;;BA)(A;OICI;FA;;;OW)",
```

Given the way the security descriptor is assigned by the filesystem, the security descriptor is always protected (P), and the auto-inherited (AI) flag is always cleared. So there's no need for the "PAI" control flags. At least "AI" shouldn't be there.

File: Modules/posixmodule.c [hunk 2]
```diff
@@ -5727,26 +5587,31 @@ os_mkdir_impl(PyObject *module, path_t *path, int mode, int dir_fd)
...
+            error = GetLastError();
+        }
     }
     if (!error) {
         result = CreateDirectoryW(path->wide, pSecAttr);
-        error = clearSecurityAttributes(&pSecAttr, &secAttrData);
-    } else {
-        // Ignore error from "clear" - we have a more interesting one already
-        clearSecurityAttributes(&pSecAttr, &secAttrData);
+        if (LocalFree(secAttr.lpSecurityDescriptor)) {
```
Comment (eryksun) [inline Modules/posixmodule.c]: This works, but you're always calling `LocalFree()` even if nothing was allocated. How about this?

```suggestion
        if (pSecAttr && LocalFree(secAttr.lpSecurityDescriptor)) {
```

Comment (zooba) [reply Modules/posixmodule.c]: It's fast (no-op) if passed `NULL`, but I guess being explicit will save people asking the question again in future.

**Pull request merged**